asynctest/templates.nim

```diff
@@ -1,6 +1,18 @@
 template suite*(name, body) =
+
   suite name:
 
+    # Runs before all tests in the suite
+    template before(beforeBody) {.dirty, used.} =
```

Comment:
It  might be confusing for a new user to learn what the difference between `setup` and `before` is. Perhaps we can use  the name`setupSuite`?

Also, I don't think this template is supposed to be dirty, because then it will leak the `b` proc into the scope of the caller. 

```suggestion
    template setupSuite(beforeBody) {.used.} =
```

Reply:
Yes, i also thought the naming might be confusing. I still think that having `setupSuite` and `setup` is still a little bit confusing. I've made the requested changes for now, but what do you think about changing to `setupSuite`, `setupTest`, `teardownTest`, `teardownSuite`?

Reply:
We could create an alias for `setupTest` and `teardownTest` -- `setup` and `teardown` that show deprecation notices.

Reply:
I'm going to go with option 2 for now, because we need this upstream 😄  Can of course change it again, no worries.

Reply:
I'd like to keep the names `setup` and `teardown`, even though all the options that you mentioned have better naming. The main reason is that I want asynctest to be a painless upgrade from `std/unittest`. Your suggestion to keep the names but add deprecation notices still adds friction when you're coming from `std/unittest` and just want to add support for async testing.

Reply:
Yeah, I agree lets keep the names as is, for the new hooks I would go with `setupAll/teardownAll`

Reply:
Updated to `setupAll`, `setup`, `teardown`, `teardownAll`.